SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc

```diff
@@ -516,13 +516,14 @@ void HcalTriggerPrimitiveAlgo::analyzeQIE11(IntegerCaloSamples& samples,
       int sampleTSminus1 = samples[ibin];
 
       if (fix_saturation_ && (sample_saturation.size() > ibin + 1))
-        check_sat = (sample_saturation[ibin + 1] || (sampleTS >= QIE11_MAX_LINEARIZATION_ET) ||
-                     sample_saturation[ibin] || (sampleTSminus1 >= QIE11_MAX_LINEARIZATION_ET));
+        check_sat = (check_sat | sample_saturation[ibin + 1] | (sampleTS > QIE11_MAX_LINEARIZATION_ET));
+      //        check_sat = (sample_saturation[ibin + 1] || (sampleTS >= QIE11_MAX_LINEARIZATION_ET) ||
+      //                     sample_saturation[ibin] || (sampleTSminus1 >= QIE11_MAX_LINEARIZATION_ET));
```

Comment:
Thanks for the suggestion. This looks like a cleaner way to write this logic

Reply:
And this modification is the same as the modification I made where all time slices are checked instead of just two of them

Reply:
> And this modification is the same as the modification I made where all time slices are checked instead of just two of them

Exact! That's why I would cite it in the PR description

Reply:
Thank you @Michael-Krohn  and @perrotta , I will leave a note in the PR description